### ISSUE TITLE
Bug: [no-unused-vars] False positive when using namespace

### ISSUE THREAD

**Issue by rubenferreira97** [typescript-eslint/typescript-eslint#11793]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

Expected result

- ESLint should not report `'Database' is defined but never used` when `Database` is actually used in `src/index.ts`.

Actual result

- ESLint incorrectly reports: `"'Database' is defined but never used.eslint@typescript-eslint/no-unused-vars"`.

### Reproduction Repository Link

https://github.com/rubenferreira97/eslint-typescript-bug

### Repro Steps

1. Install dependencies:

```cmd
bun install
```

2. Run ESLint script (`lint` : `bunx --bun eslint`):

```cmd
bun lint
```

Expected result

- ESLint should not report `'Database' is defined but never used` when `Database` is actually used in `src/index.ts`.

Actual result

- ESLint incorrectly reports: `"'Database' is defined but never used.eslint@typescript-eslint/no-unused-vars"`.

### Versions

    "eslint": "^9.39.1",
    "eslint-config-prettier": "^10.1.8",
    "eslint-plugin-import": "^2.32.0",
    "typescript-eslint": "^8.46.4"

**Comment by bradzacher** [comment]
Note for next time - please try creating a repro using our playground first, instead of creating an entire repo. You could have saved yourself a lot of time and effort!

[playground reproduction](https://typescript-eslint.io/play/#ts=5.9.2&fileType=.tsx&code=HYQwtgpgzgDiDGEAEAREAXEAjEVkG8AoJJCADxgHsAndJdATxmQBVsAbCAHhYD4kAvEhYBuQgF9ChclVpIAZgFdg8dAEtKwetHRc0mHHgA0wjsnLoIwACZRUGbLggA6Nlk56HhiL14AKYiRrLycALnsDJyNAg05wt05CAEpw-CCQvHD9RzwRejN4syRxJCISagh0RWotNODI43z3ZHExSSA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtAG4BDaKgyRE0aB2iRwYAL4hlQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

**Comment by bradzacher** [comment]
Duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/10746

**Comment by bradzacher** [comment]
Workaround: do not shadow imported names with generic names. Either rename the generic or alias the imported name.

**Comment by rubenferreira97** [comment]
> Note for next time - please try creating a repro using our playground first, instead of creating an entire repo. You could have saved yourself a lot of time and effort!
> 
> [playground reproduction](https://typescript-eslint.io/play/#ts=5.9.2&fileType=.tsx&code=HYQwtgpgzgDiDGEAEAREAXEAjEVkG8AoJJCADxgHsAndJdATxmQBVsAbCAHhYD4kAvEhYBuQgF9ChclVpIAZgFdg8dAEtKwetHRc0mHHgA0wjsnLoIwACZRUGbLggA6Nlk56HhiL14AKYiRrLycALnsDJyNAg05wt05CAEpw-CCQvHD9RzwRejN4syRxJCISagh0RWotNODI43z3ZHExSSA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtAG4BDaKgyRE0aB2iRwYAL4hlQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

Sorry about that, I genuinely thought the issue was related to using a two-file config since the error appeared in the import itself. I did try searching through GitHub first, but the search results weren’t very helpful. I’ll make sure to start with that next time.
